test/integration/backward_compatible/parallel/serialization/compact/LazyDeserialization.js

```diff
@@ -0,0 +1,206 @@
+            <serialization>
+                <compact-serialization enabled="true" />
+            </serialization>
+        </hazelcast>
+    `;
+    let SchemaNotFoundError;
+
+    try {
+        SchemaNotFoundError = require('../../../../../../lib/core/HazelcastError').SchemaNotFoundError;
+    } catch (err) {
+        // no-op
+    }
+
+    describe('ReadOnlyLazyList', function () {
+        let cluster;
+        let member;
+        let map;
+        let client;
+        let mapName;
+
+        const testFactory = new TestUtil.TestFactory();
+
+        before(async function () {
+            TestUtil.markClientVersionAtLeast(this, '5.1.0');
+            cluster = await testFactory.createClusterForParallelTests(undefined, COMPACT_ENABLED_ZERO_CONFIG_XML);
+            member = await RC.startMember(cluster.id);
+        });
+
+        beforeEach(async function() {
+            client = await testFactory.newHazelcastClientForParallelTests({
+                clusterName: cluster.id,
+            }, member);
+            mapName = TestUtil.randomString(10);
+            TestUtil.markServerVersionAtLeast(this, client, '5.1.0');
+            await putSingleCompactEntryToMapByMember();
+        });
+
+        const putSingleCompactEntryToMapByMember = async () => {
+            const script = `
+                // This class exists in remote controller.
+                var EmployeeDTO = Java.type('example.serialization.EmployeeDTO');
+                var map = instance_0.getMap("${mapName}");
+                map.set(1.0, new EmployeeDTO(1, 1));
+            `;
+            await RC.executeOnController(cluster.id, script, Lang.JAVASCRIPT);
+        };
+
+        after(async function () {
+            await testFactory.shutdownAll();
+        });
+
+        afterEach(async function() {
+            await map.destroy();
+            await testFactory.shutdownAllClients();
+            sandbox.restore();
+        });
+
+        const testReadOnlyLazyListFn = (fn) => {
+            it('should throw HazelcastSerializationError if compact schema is not known by the client', async function () {
+                map = await client.getMap(mapName);
+                const readOnlyLazyList = await map.values();
+                const error = TestUtil.getThrownErrorOrThrow(fn.bind(fn, readOnlyLazyList));
+                error.should.be.instanceOf(HazelcastSerializationError);
+                error.cause.should.be.instanceOf(SchemaNotFoundError);
+            });
+
+            it('should throw same error if any error other than HazelcastSerializationError is thrown', async function () {
+                map = await client.getMap(mapName);
+                const readOnlyLazyList = await map.values();
+                const error = TestUtil.getThrownErrorOrThrow(fn.bind(fn, readOnlyLazyList));
+                error.should.be.instanceOf(HazelcastSerializationError);
+                error.cause.should.be.instanceOf(SchemaNotFoundError);
```

Comment:
It is `HazelcastSerializationException` in java but we made it different in async clients because we need to be able to differentiate the error from other serialization exceptions. 

https://github.com/hazelcast/hazelcast/blob/cc26879d04c9e1f5db80d0fb7500ab37875f32db/hazelcast/src/main/java/com/hazelcast/internal/serialization/impl/compact/CompactStreamSerializer.java#L207

> Maybe we need to add this to the comparison spreadsheet?

I agree